Assets/Scripts/Traps/ProyectileTrap.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ProyectileTrap : MonoBehaviour
+{
+
+    [field: SerializeField] public float Damage { get; private set; }
+    [field: SerializeField] public float Speed { get; private set; }
+    [field: SerializeField] public float LifeTime { get; private set; }
+
+    [SerializeField] bool _shootReady;
+    [SerializeField] float _shootQuantity;
+
+    [Header("References")]
+
+    [SerializeField] Transform _bulletSpawnPoint;
+
+    [SerializeField] GameObject _proyectile;
```

Comment:
I'd name this field `projectilePrefab` so any person would know to put a prefab in here instead of a scene object.

Reply:
Renamed this field to `trapProjectilePrefab` as suggested.